ts/Grid/Core/Grid.ts

```diff
@@ -312,33 +312,29 @@ class Grid {
         afterLoadCallback?: (grid: Grid) => void
     ) {
         this.loadUserOptions(options);
-
-        this.querying = new QueryingController(this);
         this.id = this.options?.id || U.uniqueKey();
-
-        this.initContainers(renderTo);
-        this.initAccessibility();
-        this.initPagination();
-        this.loadDataTable(this.options?.dataTable);
-
+        this.querying = new QueryingController(this);
         this.locale = this.options?.lang?.locale || (
             (this.container?.closest('[lang]') as HTMLElement|null)?.lang
         );
-
         this.time = new TimeBase(extend<TimeBase.TimeOptions>(
             this.options?.time,
             { locale: this.locale }
         ), this.options?.lang);
 
+        fireEvent(this, 'beforeLoad');
+
+        Grid.grids.push(this);
+        this.initContainers(renderTo);
+        this.initAccessibility();
+        this.initPagination();
+        this.loadDataTable(this.options?.dataTable);
```

Comment:
Do we need this argument, in the `loadDataTable` you have an access to `this.options.dataTable`